lib/discover/local.go

```diff
@@ -154,13 +145,22 @@ func (c *localClient) sendLocalAnnouncements() {
 		select {
 		case <-c.localBcastTick:
 		case <-c.forcedBcastTick:
+		case <-stop:
+			return
 		}
 	}
 }
 
-func (c *localClient) recvAnnouncements(b beacon.Interface) {
+func (c *localClient) recvAnnouncements(stop chan struct{}) {
+	b := c.beacon
 	warnedAbout := make(map[string]bool)
 	for {
+		select {
+		case <-stop:
+			return
+		default:
+		}
+
```

Comment:
I think this will still leak as Recv blocks forever

Reply:
Not anymore

Reply:
I didn't follow all the breadcrumbs but stopping the service should at some point close the socket which unblocks Recv and results in shutdown, I hope?

Reply:
Recv reads of a channel, so connection closing won't do much.

Reply:
Recv now unblocks when stopped: https://github.com/syncthing/syncthing/pull/5950/files/f66c15d355b2b55f60b706e397839172c18c232b#diff-9564247335505ac56ab73a0d2b2310a9R92